### ISSUE TITLE
REMOTE_ADDR is incorrectly populated

### ISSUE THREAD

**Issue by oniric85** [SpiderLabs/ModSecurity#734]
Hi, all
I was trying to determine what was not working with my GeoIP database that I've set up for ModSecurity under IIS. It seems to me that REMOTE_ADDR variable is filled with the ip address concatenated to the REMOTE_PORT like

127.0.0.1:54321

This is not correct as it then causes the Geo lookup to fail as per this log line:
    GEO: Looking up "xxx.xxx.xxx.xxx:63565".
    Geo lookup for "xxx.xxx.xxx.xxx:63565" failed: The requested name is valid, but no data of the requested type was found.

I think this is a big issue as it seems not to be local to the Geo IP section but a global problem.

**Comment by ojasp** [comment]
Just ran into the same issue, I have a custom rule that blocks IP addresses based on requested URIs & it only blocks IP/port combination(192.168.0.1:40987 etc) which defeats the purpose. The lua script above doesn't seem to set the "tx.ipadres" variable. Has this been fixed in the new version? Is there another workaround?

**Comment by zimmerle** [comment]
Hi @ojasp this issue still open. There is no fix available yet. Most likely it will be fixed when the IIS add-on starts to use the libmodsecurity instead of the Standalone version.

**Comment by ojasp** [comment]
Thanks zimmerle! Do you suggest any workaround to extract IP address from remote_addr?

**Comment by ojasp** [comment]
I would like to post an update as I was able to get the above script to work after quite a lot of trial and error. When you call the lua script, please make sure to include phase & t:none. Here's my modified lua script:

```lua
#!/opt/local/bin/lua

function main()

local rawipaddress = m.getvar("REMOTE_ADDR", "none");

local ipaddress = "none";

for token in string.gmatch(rawipaddress, "[^:]+") do
    m.log(4, "IPAddress is " .. token);
    ipaddress = token
    break
end

m.setvar("TX.IPADRES", ipaddress);
m.log(4, "IP is in TX.IPADRES: " .. m.getvar("TX.IPADRES"));
return nil;
end
```
Here is how I call the lua script:
```yaml
SecRuleScript "set_ip.lua" "phase:1,t:none,pass"
SecAction phase:1,nolog,pass,initcol:user=%{TX.IPADRES},id:100
```

**Comment by zimmerle** [comment]
@ojasp good solution!

**Comment by itrade-it** [comment]
@ojasp can you explain if there is new edition of your solution?

I put this in my modsecurity_crs_15_customrules.conf

SecRuleScript "set_ip.lua" "phase:1,t:none,pass"
SecAction phase:1,nolog,pass,initcol:user=%{TX.IPADRES},id:000009
SecRule TX:IPADRES "@geoLookup" "chain,phase:1,t:none,block,log,msg:'Client IP from top 12 High Risk Countries',id:000008"
SecRule GEO:COUNTRY_CODE "@pm UA ID YU LT EG RO BG TR RU PK MY IL"

Is this correct ?

**Comment by ojasp** [comment]
That looks about right, make sure that set_ip.lua script is in base_rules
directory. Unfortunately I wouldn't be able to help you much on this
because we gave up on modsecurity early this year as it was getting too
complicated & required more time to get things ironed out than we
originally anticipated. We figured that we could do similar stuff on IIS
with Request FIltering and it didn't require as much tinkering.

Are you are running .NET/MVC application with global.asax configured to
send unhandled errors? If so, even after mod-security blocks a request,
global.asax will still send you a notification. This is not the case with
Request Filtering (maybe because Request Filtering is baked into IIS).

Anyhow, I still have my lua script and custom rule files. Let me know if
you want me to share them with you.

**Comment by itrade-it** [comment]
Thank you. 
Can you explain how do similar staff with Request Filtering ?
( getting too complicated with MS ).

> That looks about right, make sure that set_ip.lua script is in base_rules
> directory. Unfortunately I wouldn't be able to help you much on this
> because we gave up on modsecurity early this year as it was getting too
> complicated & required more time to get things ironed out than we
> originally anticipated. We figured that we could do similar stuff on IIS
> with Request FIltering and it didn't require as much tinkering.
> 
> Are you are running .NET/MVC application with global.asax configured to
> send unhandled errors? If so, even after mod-security blocks a request,
> global.asax will still send you a notification. This is not the case with
> Request Filtering (maybe because Request Filtering is baked into IIS).
> 
> Anyhow, I still have my lua script and custom rule files. Let me know if
> you want me to share them with you.
> 
> On Tue, Apr 26, 2016 at 4:58 AM, itrade-it <notifications@github.com>
wrote:
> 
> > @ojasp <https://github.com/ojasp> can you explain if there is new
edition
> > of your solution?
> >
> > I put this in my modsecurity_crs_15_customrules.conf
> >
> > SecRuleScript "set_ip.lua" "phase:1,t:none,pass"
> > SecAction phase:1,nolog,pass,initcol:user=%{TX.IPADRES},id:000009
> > SecRule TX:IPADRES "@geoLookup"
> > "chain,phase:1

**Comment by zimmerle** [comment]
Fixed by: #1220.
